docs/source/reference/literate.rst

```diff
@@ -25,35 +26,133 @@ But more importantly, a more intelligent processing of literate documents using
 Other lines are treated as documentation.
 
-We treat files with an extension of ``.lidr`` as bird style literate files.
+
+
+.. note::
+   We have diverged from ``lhs2tex`` in which ``<`` is traditionally used to display inactive code.
+   Bird-style is presented as is, and we recommended use of the other styles for much more comprehensive literate mode.
 
 Embedding in Markdown-like documents
 ====================================
 
 While Bird Style literate mode is useful, it does not lend itself well
 to more modern markdown-like notations such as Org-Mode and CommonMark.
-
 Idris2 also provides support for recognising both 'visible' and 'invisible'
-code blocks and lines in both CommonMark and OrgMode documents.
+code blocks and lines in both CommonMark and OrgMode documents using native code blocks and lines..
+
+The idea being is that:
 
-'Visible' content will be kept in the pretty printer's output while
-the 'invisible' blocks and lines will be removed.
+1. **Visible** content will be kept in the pretty printer's output;
+2. **Invisible** content will be removed; and
+3. **Specifications** will be displayed *as is* and not touched by the compiler.
 
 OrgMode
 *******
 
-+ Org mode source blocks for idris are recognised as visible code blocks,
-+ Comment blocks that begin with ``#+COMMENT: idris`` are treated as invisible code blocks.
-+ Invisible code lines are denoted with ``#+IDRIS:``.
+We treat files with an extension of ``.org`` as org-style literate files.
+Each of the following markup is recognised regardless of case:
 
-We treat files with an extension of ``.org`` as org-mode style literate files.
++ Org mode source blocks for idris sans options are recognised as visible code blocks::
 
-CommonMark
-************
+    #+begin_src idris
+    data Nat = Z | S Nat
+    #+end_src
+
++ Comment blocks that begin with ``#+BEGIN_COMMENT idris`` are treated as invisible code blocks::
+
+    #+begin_comment idris
+    data Nat = Z | S Nat
+    #+end_comment
 
-Only code blocks denoted by standard code blocks labelled as idris are recognised.
++ Visible code lines, and specifications, are not supported. Invisible code lines are denoted with ``#+IDRIS:``::
+
+    #+IDRIS: data Nat = Z | S Nat
+
++ Specifications can be given using OrgModes plain source or example blocks::
+
+    #+begin_src
+    map : (f : a -> b)
+       -> List a
+       -> List b
+    map f _ = Nil
+    #+end_src
+
+CommonMark
+**********
 
 We treat files with an extension of ``.md`` and ``.markdown`` as CommonMark style literate files.
+
++ CommonMark source blocks for idris sans options are recognised as visible code blocks::
+
+    ```idris
+    data Nat = Z | S Nat
+    ```
+
+    ~~~idris
+    data Nat = Z | S Nat
+    ~~~
+
++ Comment blocks of the form ``<!-- idris\n ... \n -->`` are treated as invisible code blocks::
+
+    <!-- idris
+    data Nat = Z | S Nat
+    -->
```

Comment:
TIL in HTML comment the closing delimiter is `-->` and not the `!-->` I have been
using for years.